source/application/SXRMB/SXRMBAppController.cpp

```diff
@@ -459,18 +407,14 @@ void SXRMBAppController::onCurrentScanActionFinishedImplementation(AMScanAction
 void SXRMBAppController::onUserConfigurationLoadedFromDb()
 {
 	AMXRFDetector *detector = SXRMBBeamline::sxrmb()->brukerDetector();
-
-	foreach (AMRegionOfInterest *region, userConfiguration_->regionsOfInterest()){
-		detector->addRegionOfInterest(region->createCopy());
-		microProbe2DScanConfiguration_->addRegionOfInterest(region);
-		exafsScanConfiguration_->addRegionOfInterest(region);
-		microProbe2DOxidationScanConfiguration_->addRegionOfInterest(region);
-	}
-
 	// This is connected here because we want to listen to the detectors for updates, but don't want to double add regions on startup.
 	connect(detector, SIGNAL(addedRegionOfInterest(AMRegionOfInterest*)), this, SLOT(onRegionOfInterestAdded(AMRegionOfInterest*)));
 	connect(detector, SIGNAL(removedRegionOfInterest(AMRegionOfInterest*)), this, SLOT(onRegionOfInterestRemoved(AMRegionOfInterest*)));
 	connect(detector, SIGNAL(regionOfInterestBoundingRangeChanged(AMRegionOfInterest*)), this, SLOT(onRegionOfInterestBoundingRangeChanged(AMRegionOfInterest*)));
+
+	foreach (AMRegionOfInterest *region, userConfiguration_->regionsOfInterest()){
```

Comment:
oh... I misunderstood your question here.

function `detector->addRegionOfInterest(region->createCopy());` will emit signal `emit addedRegionOfInterest(newRegionOfInterest);`.
SXRMBAppController is listening to that signal (Ln411):
`connect(detector, SIGNAL(addedRegionOfInterest(AMRegionOfInterest*)), this, SLOT(onRegionOfInterestAdded(AMRegionOfInterest*)));`

and in slot `onRegionOfInterestAdded`, we added the ROIs to the userConfiguration, and scanConfiguraitons.